src/components/v2/V2Project/V2ManageTokensSection/V2RedeemModal.tsx

```diff
@@ -0,0 +1,271 @@
+import { t, Trans } from '@lingui/macro'
+import { Modal, Form, Space } from 'antd'
+import { useForm } from 'antd/lib/form/Form'
+
+import InputAccessoryButton from 'components/shared/InputAccessoryButton'
+import FormattedNumberInput from 'components/shared/inputs/FormattedNumberInput'
+
+import { CSSProperties, useContext, useState } from 'react'
+import {
+  formatPercent,
+  formattedNum,
+  formatWad,
+  fromWad,
+  parseWad,
+} from 'utils/formatNumber'
+
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+import { tokenSymbolText } from 'utils/tokenSymbolText'
+import useTotalBalanceOf from 'hooks/v2/contractReader/TotalBalanceOf'
+import { NetworkContext } from 'contexts/networkContext'
+import { ThemeContext } from 'contexts/themeContext'
+import { decodeV2FundingCycleMetadata } from 'utils/v2/fundingCycle'
+import { formatRedemptionRate } from 'utils/v2/math'
+import CurrencySymbol from 'components/shared/CurrencySymbol'
+import { useETHReceivedFromTokens } from 'hooks/v2/contractReader/ETHReceivedFromTokens'
+import { V2_CURRENCY_USD } from 'utils/v2/currency'
+import { useRedeemTokensTx } from 'hooks/v2/transactor/RedeemTokensTx'
+
+// This double as the 'Redeem' and 'Burn' modal depending on if project has overflow
+export default function V2RedeemModal({
```

Comment:
Kinda sucky that we have to totally duplicate this from V1. I think you made the right decision for now though 👍